### PR TITLE
Pok 18 pok details

### LINKED ISSUE #18: POK-18- 🍿 Feature - feat: Add `PokemonDetails` functionality
# Add `PokemonDetails` functionality

Image

_Pictured, the `PokemonDetails` section in multiple states_

This story adds the ability to see details about the currently selected Pokemon. 

### States

The `PokemonDetails` section can is rendered in 2 different states: 

The first is when a `Pokemon` is selected and there are details to present. The second is the initial state when the application loads. At load there are no details to render and a message should be shown that no details are available.

### Elements
The details should be presented using two specific HTML elements. The "Stats" section should be rendered as a `<table>` component with a single row. The "Moves" section should be an un-ordered list. 

### Truncating Moves
The number of moves should be limited to 5 entries. If a Pokemon has more than five moves, move > 5 should be excluded.

### PR DESCRIPTION
**Pull request by mathiuskitchens**
# Description

<<_Please include a summary of the change._>>

## Type of change

- [ ] - fix: - patches a bug in the codebase
- [x] - feat: - introduces a new feature
- [ ] - build: - relates to build tooling / project scaffolding
- [ ] - chore: - non-code related changes, like adding packages / libraries
- [ ] - docs: - changes related to documentation

## Related Issue

Does this change have a related issue in the project?

- [x] - Yes - Closes #18 
- [ ] - No

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -14,16 +14,53 @@
...
+      <ul class="poke-moves">
+        <li id="poke-move-1">Pay-Day</li>
+        <li id="poke-move-2">Double Slap</li>
+        <li id="poke-move-3">Mega-Punch</li>
+        <li id="poke-move-4">Fire-Punch</li>
+        <li id="poke-move-5">Ice Punch</li>
+      </ul>
+    </section>
+
+    <script src="https://cdn.jsdelivr.net/npm/axios/dist/axios.min.js"></script>
```
Comment (wijohnst) [inline src/index.html:67]: Is this script supposed to be in twice?

File: src/index.js [hunk 2]
```diff
@@ -1,12 +1,66 @@
-if (process.env.NODE_ENV === 'development') {
-  const { worker } = require('./mocks/browser');
-  worker.start();
-}
+/* ß */
```
Comment (wijohnst) [inline src/index.js]: Remove?

File: src/index.js [hunk 3]
```diff
@@ -1,12 +1,66 @@
...
 function getPokemon(name) {
   axios
     .get('https://pokeapi.co/api/v2/pokemon/' + name)
-    .then((res) => console.log(res.data))
+    .then((res) => {
+      //removes blank details element
+      blankDetails.style.display = 'none';
+
+      //grabs first five moves of pokemon into an array
+      let pokemonMoves = res.data.moves
```
Comment (wijohnst) [inline src/index.js:34]: Nice.

File: src/index.js [hunk 4]
```diff
@@ -1,12 +1,66 @@
...
+
+      //Add height to table
+      let pokeHeightValue = res.data.height;
+      let HeightInTable = document.getElementById('poke-height');
+      HeightInTable.innerHTML = pokeHeightValue + '"';
+
+      //Add weight to table
+      let pokeWeightValue = res.data.weight;
+      let WeightinTable = document.getElementById('poke-weight');
+      WeightinTable.innerHTML = pokeWeightValue + ' kg';
```
Comment (wijohnst) [inline src/index.js]: Not something you have to change, but you could consider switching from string concatenation to string interpolation.
```suggestion
      WeightinTable.innerHTML = `${pokeWeightValue} kg`;
```

File: src/index.js [hunk 5]
```diff
@@ -1,12 +1,66 @@
...
+        .map((move) => move.move.name);
+
+      //Add height to table
+      let pokeHeightValue = res.data.height;
+      let HeightInTable = document.getElementById('poke-height');
+      HeightInTable.innerHTML = pokeHeightValue + '"';
+
+      //Add weight to table
+      let pokeWeightValue = res.data.weight;
+      let WeightinTable = document.getElementById('poke-weight');
```
Comment (wijohnst) [inline src/index.js]: The convention is to name variables using camel-case. 

```suggestion
      let weightInTable = document.getElementById('poke-weight');
```

File: src/index.js [hunk 6]
```diff
@@ -1,12 +1,66 @@
...
+
+      //Add move4 to DOM
+      let pokeMove4 = pokemonMoves[3];
+      let ListMove4 = document.getElementById('poke-move-4');
+      ListMove4.innerHTML = pokeMove4;
+
+      //Add move5 to DOM
+      let pokeMove5 = pokemonMoves[4];
+      let ListMove5 = document.getElementById('poke-move-5');
+      ListMove5.innerHTML = pokeMove5;
```
Comment (wijohnst) [inline src/index.js:76]: So, you mentioned that you were trying to figure out how to do all of these DOM updates "with a function". That idea, of reducing a process down to its generic, repeatable steps, is what we call an [abstraction](https://en.wikipedia.org/wiki/Abstraction_(computer_science)). 

And your impulse was correct, there is a way to "abstract" these repetitive and predictable DOM manipulations. Abstractions like that make up the core of what a framework like React does. 

We could talk about refactoring this at some point but for now its great. Definitely tedious, but great.

File: src/styles.css [hunk 7]
```diff
@@ -180,3 +180,62 @@ body {
   border: 0;
   margin-left: 10px;
 }
+
+#empty-pokemon-details {
+  box-sizing: border-box;
```
Comment (wijohnst) [inline src/styles.css]: If you find yourself repeating similar styles, you can move them into a higher scope (this is the _cascading_ in _Cascading Style Sheets_). So if you want to use `box-sizing: border-box` everywhere, you can add it to the `html` and `body` selectors:

```css
html, body {
  box-sizing: border-box;
}
```

**Review by wijohnst — COMMENTED** [review]
Looks good. Left a few comments.

File: src/index.js (hunk 2 above)
Comment (mathiuskitchens) [reply src/index.js]: Removed

File: src/index.html (hunk 1 above)
Comment (mathiuskitchens) [reply src/index.html:67]: Nope!

File: src/index.js (hunk 4 above)
Comment (mathiuskitchens) [reply src/index.js]: Agree, forgot about that method, updated

File: src/index.js (hunk 5 above)
Comment (mathiuskitchens) [reply src/index.js]: updated

File: src/index.js (hunk 6 above)
Comment (mathiuskitchens) [reply src/index.js:76]: Awesome, thanks!

File: src/styles.css (hunk 7 above)
Comment (mathiuskitchens) [reply src/styles.css]: Still learning best practices here, thanks!

**Review by wijohnst — APPROVED** [review]
Let's merge it!

**Review by wijohnst — APPROVED** [review]
Looks good to me!

**Pull request merged**